homeassistant/components/velux/light.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Velux lights."""
+from pyvlx import Intensity, LighteningDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.core import callback
+
+from . import DATA_VELUX
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up light(s) for Velux platform."""
+    entities = []
+    for node in hass.data[DATA_VELUX].pyvlx.nodes:
+        if isinstance(node, LighteningDevice):
+            entities.append(VeluxLight(node))
+    async_add_entities(entities)
+
+
+class VeluxLight(LightEntity):
+    """Representation of a Velux light."""
+
+    def __init__(self, node):
+        """Initialize the light."""
+        self.node = node
+
+    @callback
+    def async_register_callbacks(self):
+        """Register callbacks to update hass after device was changed."""
+
+        async def after_update_callback(device):
+            """Call after device was updated."""
+            self.async_write_ha_state()
+
+        self.node.register_device_updated_cb(after_update_callback)
+
+    async def async_added_to_hass(self):
+        """Store register state change callback."""
+        self.async_register_callbacks()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this cover."""
+        return self.node.serial_number
+
+    @property
+    def name(self):
+        """Return the name of the Velux device."""
+        if not self.node.name:
+            return "Light #" + str(self.node.node_id)
+        return self.node.name
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
```

Comment:
Looking at https://github.com/home-assistant/core/pull/47720, it seems that SUPPORT_BRIGHTNESS is deprecated.

Reply:
Good catch, will change this asap

Reply:
Did have a look into changing this but this give me some problems.

- In the frontend (lovelace cards) there is no option for the brightness slider when supported_color_mode is brightness
- The return type `-> set | None` gives me errors `TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'`.
- No other light component implements this method yet. Wanted to have a look if it works correctly on other components.

Seems this is a feature which is still in kinda alpha state.

I don't have that much experience in Python programming yet and knowledge about the home assistant core to deep dive in this problem.

Reply:
I have left the supported_features `SUPPORT_BRIGHTNESS` in place for now as this is working correctly in the dashboard card.

Maybe a core member could elaborate why the following (non deprecated approach) is not working yet:

```
def supported_color_modes(self):
        """Flag supported color modes."""
        return {COLOR_MODE_BRIGHTNESS}
```

Reply:
You probably need to rebase this PR in order to get correct color mode support.
But @iMicknl is correct, this needs to be replaced with color modes.

Reply:
This has been resolved. Implemented supported color modes and remove supported_features. Also implemented by setting attributes as this looks like the new preferred way.